lib/process-services/src/lib/process-list/components/start-process.component.spec.ts

```diff
@@ -747,4 +709,104 @@ describe('StartFormComponent', () => {
+        });
+
+        it('should show no process definition available template if processDefinitions are empty', () => {
+            const change = new SimpleChange(null, 3, true);
+            component.ngOnChanges({ 'appId': change });
+            component.processDefinitions = [];
+            fixture.detectChanges();
+            const noProcessElement = fixture.nativeElement.querySelector('.adf-empty-content__title');
+            expect(noProcessElement).not.toBeNull('Expected no available process message to be present');
+            expect(noProcessElement.innerText.trim()).toBe('ADF_PROCESS_LIST.START_PROCESS.NO_PROCESS_DEFINITIONS');
+        });
+
+        it('should show no process definition selected template if there is no process definition selected', () => {
+            component.showSelectApplicationDropdown = true;
+            const change = new SimpleChange(null, 3, true);
+            component.ngOnChanges({ 'appId': change });
+            fixture.detectChanges();
+            const noProcessElement = fixture.nativeElement.querySelector('.adf-empty-content__title');
+            expect(noProcessElement).not.toBeNull('Expected no available process message to be present');
+            expect(noProcessElement.innerText.trim()).toBe('ADF_PROCESS_LIST.START_PROCESS.NO_PROCESS_DEF_SELECTED');
+        });
+
+        it('should show no start form template if selected process definition does not have start form', () => {
+            component.showSelectApplicationDropdown = true;
+            const change = new SimpleChange(null, 3, true);
+            component.ngOnChanges({ 'appId': change });
+            component.selectedProcessDef = testMultipleProcessDefs[0];
+            fixture.detectChanges();
+            const noProcessElement = fixture.nativeElement.querySelector('.adf-empty-content__title');
+            expect(noProcessElement).not.toBeNull('Expected no available process message to be present');
+            expect(noProcessElement.innerText.trim()).toBe('ADF_PROCESS_LIST.START_PROCESS.NO_START_FORM');
+        });
+    });
+
+    describe('Error event', () => {
+
+        const processDefError = { message: 'Failed to load Process definitions' };
+        const applicationsError = { message: 'Failed to load applications' };
+        const startProcessError = { message: 'Failed to start process' };
+
+        beforeEach(() => {
+            fixture.detectChanges();
+        });
+
+        it('should emit error event in case loading process definitions failed', async(() => {
+            getDefinitionsSpy.and.returnValue(throwError(processDefError));
+            const errorSpy = spyOn(component.error, 'emit');
+            const change = new SimpleChange(null, 3, true);
+            component.ngOnChanges({ 'appId': change });
+            fixture.detectChanges();
+            fixture.whenStable().then(() => {
+                expect(errorSpy).toHaveBeenCalledWith(processDefError);
+            });
```

Comment:
can we use await fixture.whenStable(); And then the expectation in the next line?